### ISSUE TITLE
Found with fuzzer: obfree: deleting worn obj (318: 1024)

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#25]
```

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7363859 in __GI_abort () at abort.c:79
#2  0x0000555555a44eea in NH_abort () at end.c:236
#3  0x0000555555a4bce6 in panic (str=0x555556524320 "%s") at end.c:802
#4  0x0000555555f99d55 in impossible (s=0x555556555b60 "obfree: deleting worn obj (%d: %ld)")
    at pline.c:518
#5  0x00005555560b367f in obfree (obj=0x60c000138700, merge=0x0) at shk.c:1152
#6  0x0000555555af6127 in delobj_core (obj=0x60c000138700, force=0 '\000') at invent.c:1264
#7  0x0000555555af5f61 in delobj (obj=0x60c000138700) at invent.c:1238
#8  0x0000555555eeb072 in do_play_instrument (instr=0x60c000138700) at music.c:1016
#9  0x00005555557b9f13 in doapply () at apply.c:4973
#10 0x000055555586306f in rhack (cmd=0x5555567a5800 <in_line> "a") at cmd.c:5517
#11 0x0000555555763c00 in moveloop (resuming=0 '\000') at allmain.c:798
#12 0x00005555563bbbdc in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 0, 0, 0, 0, 140737335783280, 93824997411772, 335544320, 
            140737341046928, 1073741824, 0, 140737342793376, 4294967295, 140737337360384, 
            140737342776480, 3264981123307395840}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff7363859 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0xffffffff, sa_sigaction = 0xffffffff}, 
          sa_mask = {__val = {140737337360384, 140737342776480, 3264981123307395840, 
              140737488344192, 140737342793376, 18446744073709547520, 140737327104000, 
              140737345462488, 140737488345264, 140737488345264, 140737344331794, 
              17592186043024, 140737488338944, 140737341046928, 3264981123307395840, 
              140737488344592}}, sa_flags = 1964497664, sa_restorer = 0x7fffffffd7d0}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x0000555555a44eea in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#3  0x0000555555a4bce6 in panic (str=0x555556524320 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fffffffd8a0, 
            reg_save_area = 0x7fffffffd7d0}}
#4  0x0000555555f99d55 in impossible (s=0x555556555b60 "obfree: deleting worn obj (%d: %ld)")
    at pline.c:518
        pbuf = "obfree: deleting worn obj (318: 1024)\000\000\000\020\333\377\377\377\177\000\000\070\233\003\000 a\000\000\354h8\367\377\177\000\000\336\362\256UUU\000\000\030\000\000\000\000\000\000\000 \233\003\000 a\000\000\000\333\027u\b\213O-\300\232\003\000 a\000\000<\373\377\377\377\017\000\000\340\331\377\377\377\177\000\000\240\333\377\377\377\177\000\000\340\331\377\377\377\177\000\000\240\342\377\377\377\177\000\000\300\333\377\377\377\177\000\000\001\356\206UUU\000\000\000\000\000\000\000\000\000\000\030", '\000' <repeats 15 times>, "\340\334\377\377\377\177\000\000\200\325\016\000\300`\000\000\000\000\000"...
        the_args = {{gp_offset = 24, fp_offset = 48, overflow_arg_area = 0x7fffffffdf90, 
            reg_save_area = 0x7fffffffdeb0}}
#5  0x00005555560b367f in obfree (obj=0x60c000138700, merge=0x0) at shk.c:1152
        bp = 0x0
        bpm = 0x7fffffffe400
        shkp = 0x0
#6  0x0000555555af6127 in delobj_core (obj=0x60c000138700, force=0 '\000') at invent.c:1264
        update_map = 0 '\000'
#7  0x0000555555af5f61 in delobj (obj=0x60c000138700) at invent.c:1238
No locals.
#8  0x0000555555eeb072 in do_play_instrument (instr=0x60c000138700) at music.c:1016
        buf = '\000' <repeats 255 times>
        c = 121 'y'
        s = 0x7fffffffe080 "\263\212\265A"
        x = 1279744
        y = 1102416563
        ok = 0 '\000'
        instr_breaks = 1 '\001'
#9  0x00005555557b9f13 in doapply () at apply.c:4973
        obj = 0x60c000138700
        otmp = 0x0
        res = 1
        class_list = "\006\002\v\000\000\000\000\000\240\240EVUU\000\000jԆU"
#10 0x000055555586306f in rhack (cmd=0x5555567a5800 <in_line> "a") at cmd.c:5517
        tlist = 0x555556664e60 <extcmdlist+192>
        res = 1450859940
        func = 0x5555557b8be8 <doapply>
        spkey = 0
        prefix_seen = 0 '\000'
        bad_command = 85 'U'
        firsttime = 1 '\001'
#11 0x0000555555763c00 in moveloop (resuming=0 '\000') at allmain.c:798
        moveamt = 24
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 27427
        past_clock = -17993
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#12 0x00005555563bbbdc in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
<img width="960" alt="issue 25" src="https://user-images.githubusercontent.com/5000096/198625453-d9de1eb7-36ae-4b64-9c9f-e01724430273.png">

**Comment by elunna** [comment]
Fixed in commit e271184c8.
